Erratically failing test: mortar/periodic_segmental_constraint.periodic_checker2d

## Bug Description
There is a test which seems to spontaneously fail during CI:

```pre
mortar/periodic_segmental_constraint.periodic_checker2d: Running csvdiff: /data/civet0/build/moose/python/mooseutils/csvdiff.py /data/civet0/build/moose/test/tests/mortar/periodic_segmental_constraint/gold/periodic_checker2d_out.csv /data/civet0/build/moose/test/tests/mortar/periodic_segmental_constraint/periodic_checker2d_out.csv --relative-tolerance 5.5e-06 --abs-zero 1e-10
mortar/periodic_segmental_constraint.periodic_checker2d: ERROR: In file periodic_checker2d_out.csv: Columns with header 'time' aren't the same length
mortar/periodic_segmental_constraint.periodic_checker2d: 
mortar/periodic_segmental_constraint.periodic_checker2d: ################################################################################
mortar/periodic_segmental_constraint.periodic_checker2d: Tester failed, reason: CSVDIFF
mortar/periodic_segmental_constraint.periodic_checker2d: 
[0.340s]     DIFF mortar/periodic_segmental_constraint.periodic_checker2d FAILED (CSVDIFF) [recover]
```

## Steps to Reproduce
Impossible, as this happens erratically 

## Impact
Prevents CI from continuing unabated

I haven't pulled moose changes since 5/16/2023 or so. Now when I've installed PETSC and libmesh using the scripts, and go to install the moose-test-opt, I get errors from <filesystem> missing:

/home/ttruster/_Prog_Repo/MOOSE/moose/framework/build/header_symlinks/RestartableDataIO.h:18:10: fatal error: filesystem: No such file or directory
 #include <filesystem>

I searched my older repo using grep, and those includes must have been added in the last year.
In the current repo, they are mostly in the Restart source folder.

output of g++ --version:
g++ (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Copyright (C) 2017 Free Software Foundation, Inc.

Do I need a newer installation? I'm running on Ubuntu 18.04 and have access to 20.04 on another computer. Python version is 3.something

--- a/test/tests/mortar/periodic_segmental_constraint/periodic_aux2d.i
+++ b/test/tests/mortar/periodic_segmental_constraint/periodic_aux2d.i
@@ -181,8 +181,8 @@
 
 [Executioner]
   type = Steady
-  petsc_options_iname = '-pc_type -pc_factor_shift_type'
-  petsc_options_value = 'lu       NONZERO'
+  petsc_options_iname = '-pc_type -pc_factor_shift_type -pc_factor_shift_amount'
+  petsc_options_value = 'lu       NONZERO               1e-15'
   solve_type = NEWTON
 []
 

--- a/test/tests/mortar/periodic_segmental_constraint/periodic_checker2d.i
+++ b/test/tests/mortar/periodic_segmental_constraint/periodic_checker2d.i
@@ -242,8 +242,8 @@
 
 [Executioner]
   type = Steady
-  petsc_options_iname = '-pc_type -pc_factor_shift_type'
-  petsc_options_value = 'lu       NONZERO'
+  petsc_options_iname = '-pc_type -pc_factor_shift_type -pc_factor_shift_amount'
+  petsc_options_value = 'lu       NONZERO               1e-15'
   solve_type = NEWTON
 []
 

--- a/test/tests/mortar/periodic_segmental_constraint/periodic_simple2d.i
+++ b/test/tests/mortar/periodic_segmental_constraint/periodic_simple2d.i
@@ -173,8 +173,8 @@
 
 [Executioner]
   type = Steady
-  petsc_options_iname = '-pc_type -pc_factor_shift_type'
-  petsc_options_value = 'lu       NONZERO'
+  petsc_options_iname = '-pc_type -pc_factor_shift_type -pc_factor_shift_amount'
+  petsc_options_value = 'lu       NONZERO               1e-15'
   solve_type = NEWTON
 []
 

--- a/test/tests/mortar/periodic_segmental_constraint/periodic_simple3d.i
+++ b/test/tests/mortar/periodic_segmental_constraint/periodic_simple3d.i
@@ -217,8 +217,8 @@
 
 [Executioner]
   type = Steady
-  petsc_options_iname = '-pc_type -pc_factor_shift_type'
-  petsc_options_value = 'lu       NONZERO'
+  petsc_options_iname = '-pc_type -pc_factor_shift_type -pc_factor_shift_amount'
+  petsc_options_value = 'lu       NONZERO               1e-15'
   solve_type = NEWTON
 []
 

